src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/refreshanalyzer/RefreshSearchAnalyzerResponse.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.refreshanalyzer
+
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.support.DefaultShardOperationFailedException
+import org.elasticsearch.action.support.broadcast.BroadcastResponse
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.xcontent.ConstructingObjectParser
+import org.elasticsearch.common.xcontent.ToXContent.Params
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.rest.action.RestActions
+import java.io.IOException
+import java.util.function.Function
+
+class RefreshSearchAnalyzerResponse : BroadcastResponse {
+
+    private var results: MutableMap<String, List<String>> = HashMap()
+    private var shardFailures: MutableList<FailedShardDetails> = mutableListOf()
+    private var temp: List<DefaultShardOperationFailedException> = mutableListOf()
+
+    protected var logger = LogManager.getLogger(javaClass)
+
+    @Throws(IOException::class)
+    constructor(inp: StreamInput) : super(inp) {
+        val resultSize: Int = inp.readVInt()
+        for (i in 0..resultSize) {
+            results.put(inp.readString(), inp.readStringArray().toList())
+        }
+
+        val failureSize: Int = inp.readVInt()
+        for (i in 0..failureSize) {
+            shardFailures.add(FailedShardDetails(inp.readString(), inp.readInt(), inp.readString()))
```

Comment:
nit: Make FailedShardDetails writeable and just pass in the streaminput/output

Reply:
Good point. I wanted to do some refactoring on this but couldn't get to it earlier. It's done now, the new version does not use this class now.